### ISSUE TITLE
View Layer: manual order and hide for source entries

### ISSUE THREAD

**Issue by zembutsu** [zembutsu/erimil#268]
## Summary

Add a user-editable view layer on top of sources (PDF / ZIP / folder) that allows:

1. **Manual order** — reorder entries via drag or keyboard, independent of the underlying source
2. **Hide** — mark individual entries as hidden from Grid / Viewer / Slide without modifying the source

The source file itself is never modified. All edits live in Erimil-managed sidecar metadata keyed by content hash, so edits survive source relocation.

This issue is the first concrete realization of the **Editorial Model** described in [`docs/DESIGN-PHILOSOPHY.md`](../docs/DESIGN-PHILOSOPHY.md).

## Motivation

Erimil treats sources as read-only. Users with real-world material hit walls:

- **Photographers**: want to mark rejects from a shoot without deleting them. Export to ZIP / folder of keepers.
- **Archive reviewers**: want to skip duplicate pages, ads, or inserted bonus content in fan-made archives.
- **PDF readers**: want to skip blank pages, rearrange out-of-order scans, or chapter-reorder study materials.
- **Folder curators**: want to promote a cover image or deprioritize supporting files without resorting to filename hacks (`00_cover.png`).

The common thread: existing tools force "rewrite the source" or "give up". Erimil's position of **never modifying the source** (DESIGN-PHILOSOPHY Principle 1: *Source is Sacred*) creates room for a third option — a **non-destructive view overlay**.

## Core concept: the View Layer

Today: `Source → Grid / Viewer / Slide`

Proposed: `Source → View Layer → Grid / Viewer / Slide`

The View Layer is per-source, stored separately, and applied at display time. It is an overlay, not a transform of the source.

This issue introduces the View Layer with two capabilities (manual order, hide). Future overlay capabilities (per-entry notes, tags, groupings, named arrangements) can be added to the same layer without architectural change.

## Design principles

These follow from `docs/DESIGN-PHILOSOPHY.md`:

1. **Source is sacred** — PDF / ZIP / folder contents never modified by Erimil.
2. **Metadata follows content, not path** — keyed by filename + hash, survives relocation.
3. **View Layer is per-source** — no global user-level overlay; each source has its own.
4. **Unified storage** — manual order and hide share one metadata record per source (no separate files per capability).
5. **Erimil-managed directory** — sidecar metadata lives in Erimil's own storage, not alongside sources. Works with read-only media.

## Data model (sketch)

One file per source, keyed by source hash:

```
~/Library/.../Erimil/ViewLayers/
  <source_hash>.json
```

Content:

```json
{
  "sourceHash": "...",
  "sourcePath": "last known path (debugging only, not used for lookup)",
  "updatedAt": "2026-04-25T...",
  "entries": {
    "<entry_hash_or_filename>": {
      "order": 3,
      "hidden": false
    }
  }
}
```

Notes:

- **Per-entry key**: content hash preferred; fallback to filename if hash unavailable. PDF pages need a stable per-page identifier (see Open Questions).
- **`order` is a sparse integer**, not a permutation over all entries. Entries without an `order` value fall back to the basis sort order (#267).
- **`hidden` defaults to false** — only present in the record when the user has acted.
- **Sparse storage** — entries without manual state don't appear in the JSON.

## Interaction with favorite system

| User state | favorite | hidden | Display in Grid | Export ★-only (Issue #269) |
|-----------|----------|--------|-----------------|--------------------------|
| Neutral | false | false | shown | excluded |
| Favorited | true | false | shown, ★ badge | included |
| Hidden | false | true | **not shown** | excluded |
| Favorited + Hidden | true | true | **not shown** (hide wins for display) | **included** (favorite wins for export) |

**Key rule**: `hide` suppresses display, but does **not** suppress `favorite` status. This enables the workflow of "mark favorites → hide rejects → export favorites", where favorites persist through hidden state.

A "Show hidden" toggle in management UI reveals hidden entries (greyed) so the user can unhide.

### Manual order

- **Drag in Grid**: drag entry to new position; remaining entries shift around it.
- **Keyboard (Viewer or Grid)**: candidate bindings — `Option+Shift+Up/Down` to nudge current entry in order.
- **Reset**: "Reset to default order" in context menu per source.

### Hide

- **Keyboard**: candidate binding — `H` to toggle hide on selected entry.
- **Visual**: hidden entries removed from Grid by default; visible (greyed) under a "Show hidden" toggle.
- **Batch**: multi-select + hide (leverages existing multi-selection).

### Management

- Per-source View Layer status display: entry count, hidden count, custom order yes/no.
- Reset all (clear View Layer for this source).
- Available from toolbar or context menu.

## Out of scope

- **Modifying the source file** — never. (Principle 1.)
- **Global-level View Layer** (applies across multiple sources) — out of scope; per-source only.
- **Groups / sub-albums** — future overlay capability, not this issue.
- **Per-entry notes / tags** — future overlay capability, not this issue.
- **Multiple named arrangements per source** — future, not this issue.
- **Syncing View Layers across devices** — out of scope.

## Technical considerations

- **Hash key strategy**: folder entries use file content hash; PDF pages need a stable per-page key (page index + source hash composite as a candidate — see Open Questions).
- **Entry add / remove reconciliation**: new entries appear at default position, visible. Entries missing from source but present in View Layer are GC'd lazily on next save.
- **GC timing**: avoid GC during read-only operations; only when the user modifies the layer.
- **Performance**: View Layer must be O(1) per-entry lookup during Grid rendering. Load all entries once per source into memory.
- **Crash safety**: write to temp file + atomic rename on save. Metadata loss = curation loss, which is irreversible from the user's perspective.
- **Migration from current state**: no existing data to migrate; this is new.

## Open questions

- **PDF page identity**: does Erimil have a stable per-page hash today, or only per-PDF? If only per-PDF, we need to establish per-page keying. Candidates: per-page content hash (changes if user re-OCRs); page index + source hash composite (fragile if pages are inserted in a future PDF revision the user re-opens). Both have failure modes.
- **Newly-added entries in manually-ordered source**: append at end (visible)? Insert at content-hash-preserving position? Show with a "new" indicator until the user acknowledges?
- **Multi-window**: if the same source is open in two windows, how is View Layer state shared? Likely last-write-wins; metadata is source-local, not window-local. Confirm.
- **End-user naming**: "View Layer" is the architectural term. End-user UI labels (Arrangement / Curation / 棚 / etc.) deferred — see `docs/DESIGN-PHILOSOPHY.md` Open Questions.

## Implementation plan (phased)

1. **Phase 1**: Data model + storage + load/save. No UI yet. Tests against the data model alone.
2. **Phase 2**: Hide feature end-to-end. Simpler than reorder; ships value quickly.
3. **Phase 3**: Manual order feature — drag in Grid, keyboard nudge. Depends on #267 (basis sort).
4. **Phase 4**: Management UI — show hidden toggle, reset, status display.
5. **Phase 5**: Export filtered by favorite — covered in Issue #269.

## Depends on / blocks

- **Depends on**: #267 (basis sort) for Phase 3.
- **Blocks**: Issue #269 (export ★-only) — needs hide / favorite disambiguation defined here.
